util/azure/utils.go

```diff
@@ -338,20 +407,26 @@ func fetchApiResources(settings *DiscoverResourcesSettings) ([]*metav1.APIResour
 	}
 
 	apiresourcesList, err := kubeclientset.Discovery().ServerPreferredResources()
-	if err != nil {
-		return nil, err
-	}
 
 	if klog.V(5).Enabled() {
 		printApiresourcesList, _ := json.Marshal(apiresourcesList)
 
 		klog.Infof("List of ApiResources fetched from apiserver: %s", string(printApiresourcesList))
 	}
 
+	if err != nil {
+		// ignoring unreachable apiservice error.
+		if strings.Contains(err.Error(), retrieveServerApiError) && strings.Contains(err.Error(), apiserviceUnreachableError) {
+			klog.Infof("Error while fetching apiservices from apiserver: %s", err.Error())
```

Comment:
It seems in this error case we are swallowing error and continuing with discover resource. Should we return err and skip further processing?

Reply:
don't understand , we need to skip the apiservice error , only if error is not the apiservice error then we need to return err as discussed

Reply:
Can we add UT to verify partial list scenario to make sure existing data is not removed.

Reply:
added test